test/helpers/kubectl.go

```diff
@@ -578,6 +579,72 @@ func (kub *Kubectl) WaitCleanAllTerminatingPods() error {
 	return err
 }
 
+// ApplyNetworkPolicyUsingAPI apply a Kubernetes network policy using the API
+// and waits up  until timeout seconds for the policy to be applied in all
+// Cilium endpoints. Returns an error if the policy is not imported before the
+// timeout is exceeded.
+func (kub *Kubectl) ApplyNetworkPolicyUsingAPI(namespace string, networkPolicy *networkingv1.NetworkPolicy) error {
```

Comment:
As time goes on, I am in agreement with what you said initially back when this test was merged in that we should just move to using our own wrappers around `kubectl` instead of using the K8s API directly (or we move to using the K8s APIs directly). Otherwise, we will have to support both ways of testing (CLI vs. API directly) and will hit gaps like this. I'm not saying that we should do this as part of this fix; this is more of a general comment. What do you think?

Reply:
100%, I was thinking to raise this one in the next meeting :-)